int32 data
---
bool success
int32 newData